presto-client/src/client.ts

```diff
@@ -0,0 +1,121 @@
+import { PrestoClientConfig, PrestoQuery, PrestoResponse } from './client.types'
+
+export class PrestoClient {
+  private baseUrl: string
+  private catalog?: string
+  private headers: Record<string, string>
+  private interval: number
+  private schema?: string
+  private timezone?: string
+  private user: string
+
+  constructor({ catalog, host, interval, port, schema, timezone, user }: PrestoClientConfig) {
+    this.baseUrl = `${host || 'http://localhost'}:${port || 8080}/v1/statement`
+    this.catalog = catalog
+    this.interval = interval || 100
+    this.schema = schema
+    this.timezone = timezone
+    this.user = user
```

Comment:
this reminds me of another optional param: `source` and its default value should be `presto-js-client`

Reply:
I'll add it right away

Reply:
The corresponding header is `X-Presto-Source`

Reply:
You are faster than me. I should also enclose the header name in the beginning. my bad.

Reply:
no worries! I thought the source was the client info, I'll commit the change in a minute